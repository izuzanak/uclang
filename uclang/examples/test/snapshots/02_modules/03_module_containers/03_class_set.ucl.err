 ---------------------------------------- 
Exception: ERROR: in file: "03_class_set.ucl" on line: 170
    ("set.remove(6) -> %s\n" % set.remove(6).to_string()).print();

Set does not contain requested element
 ---------------------------------------- 
