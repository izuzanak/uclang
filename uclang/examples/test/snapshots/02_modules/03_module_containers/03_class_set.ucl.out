
=== Set object =================================================================

Set: []
Set: [1,2,3,4,5]
Set: [1,2,4]

=== Set assign operator ========================================================

set: []
set: 10

=== Set compare operators ======================================================

set_0 == set_0 -> 1
set_0 == set_1 -> 0
set_0 == 10    -> 0
set_0 != set_0 -> 0
set_0 != set_1 -> 1
set_0 != 10    -> 1

=== Set plus operators =========================================================

set_0          -> [1,2,3]
arr_1          -> [5,4,2]
set_0 + arr_1  -> [1,2,3,4,5]
set_0          -> [1,2,3]
set_0 += arr_1 -> [1,2,3,4,5]
set_0          -> [1,2,3,4,5]

=== Set difference operators ===================================================

set_0          -> [1,2,3,4]
set_1          -> [3,4,5,6]
set_0 - set_1  -> [1,2]
set_1 - set_0  -> [5,6]
set_0          -> [1,2,3,4]
set_0 -= set_1 -> [1,2]
set_0          -> [1,2]

=== Set intersection operators =================================================

set_0          -> [1,2,3,4]
set_1          -> [3,4,5,6]
set_0 & set_1  -> [3,4]
set_0          -> [1,2,3,4]
set_0 &= set_1 -> [3,4]
set_0          -> [3,4]

=== Set union operators ========================================================

set_0          -> [1,2,3,4]
set_1          -> [3,4,5,6]
set_0 | set_1  -> [1,2,3,4,5,6]
set_0          -> [1,2,3,4]
set_0 |= set_1 -> [1,2,3,4,5,6]
set_0          -> [1,2,3,4,5,6]

=== Set symmetric difference operators =========================================

set_0          -> [1,2,3,4]
set_1          -> [3,4,5,6]
set_0 ^ set_1  -> [1,2,5,6]
set_0          -> [1,2,3,4]
set_0 ^= set_1 -> [1,2,5,6]
set_0          -> [1,2,5,6]

=== Set subset test operators ==================================================

set_0          -> [3,4,5]
set_1          -> [3,4,5,6]
set_0 <= set_1 -> 1
set_1 <= set_0 -> 0
set_0 >= set_1 -> 0
set_1 >= set_0 -> 1

=== Set item select operator ===================================================

set[0]: One
set[2]: Two
set[4]: Four
set[3]: Three

=== Set method clear#0 =========================================================

set         -> [1,2,3,4,5]
set.clear() -> []
set         -> []

=== Set method items#0 =========================================================

set   -> type: Set, value [1,2,3,4,5]
items -> type: Array, value [1,2,3,4,5]
set   -> type: Set, value [1,2,3,4,5]

=== Set method insert#1 ========================================================

set            -> [1,2,3,4,5]
set.insert(10) -> [1,2,3,4,5,10]
set.insert(10) -> [1,2,3,4,5,10]
set.insert(9)  -> [1,2,3,4,5,9,10]
set            -> [1,2,3,4,5,9,10]

=== Set method remove#0 ========================================================

set           -> [1,2,3,4,5]
set.remove(2) -> [1,3,4,5]
set.remove(4) -> [1,3,5]
set           -> [1,3,5]
set           -> [1,3,5]

=== Set method contain#0 =======================================================

set            -> [1,2,3,4,5]
set.contain(2) -> 1
set.contain(4) -> 1
set.contain(6) -> 0
set            -> [1,2,3,4,5]

=== Difference of to_string#0 and to_string#1 ==================================

set.to_string()        -> [1,2,3,4,5]
set.to_string("-SEP-") -> 1-SEP-2-SEP-3-SEP-4-SEP-5

=== Tutorial end ===============================================================

